challenges-09.test.js

```diff
@@ -0,0 +1,322 @@
+  gender: 'n/a'},
+{
+  name: 'R2-D2',
+  height: '96',
+  mass: '32',
+  hair_color: 'n/a',
+  skin_color: 'white, blue',
+  eye_color: 'red',
+  birth_year: '33BBY',
+  gender: 'n/a'
+},
+{
+  name: 'Darth Vader',
+  height: '202',
+  mass: '136',
+  hair_color: 'none',
+  skin_color: 'white',
+  eye_color: 'yellow',
+  birth_year: '41.9BBY',
+  gender: 'male'
+},
+{
+  name: 'Leia Organa',
+  height: '150',
+  mass: '49',
+  hair_color: 'brown',
+  skin_color: 'light',
+  eye_color: 'brown',
+  birth_year: '19BBY',
+  gender: 'female'
+}];
+
+const returnNames = (arr) => {
+//   Solution code here...
+const charNames = arr.reduce( (acc,val) => {
+    acc.push( val.name );
+    return acc ;
+  }, [] );
+  return charNames ;
+  // Solution code here...
+//   const names = arr.reduce( (acc, val, idx) => {
+//     return( val.name );
+//     return acc;
+//   }, [] );
+
+//  let name = arr.reduce( function(acc , value ,idx ){
+//      return acc   
+//      } , [])
+};
+
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 3
+
+Write a function named reversedString that takes in a string and returns a string with the letters in reverse order.
+
+Note: You must use reduce for this challenge. You may not use the built-in .reverse() string method.
+------------------------------------------------------------------------------------------------ */
+
+const reversedString = (str) => {
+  // Solution code here...
+  let newThing = str.split('')
+  let reversed = newThing.reduce ((newstring , currentlatter)=> {
+  return currentlatter + newstring ;
+}, '')
+return reversed ;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 4
+
+Write a function named countNumberOfChildren that, given the array of characters, below, uses reduce to return the total number of children in the data set.
+------------------------------------------------------------------------------------------------ */
+
+const characters = [
+  {
+    name: 'Eddard',
+    spouse: 'Catelyn',
+    children: ['Robb', 'Sansa', 'Arya', 'Bran', 'Rickon'],
+    house: 'Stark',
+  },
+  {
+    name: 'Jon',
+    spouse: 'Lysa',
+    children: ['Robin'],
+    house: 'Arryn',
+  },
+  {
+    name: 'Cersei',
+    spouse: 'Robert',
+    children: ['Joffrey', 'Myrcella', 'Tommen'],
+    house: 'Lannister',
+  },
+  {
+    name: 'Daenarys',
+    spouse: 'Khal Drogo',
+    children: ['Drogon', 'Rhaegal', 'Viserion'],
+    house: 'Targaryen',
+  },
+  {
+    name: 'Mace',
+    spouse: 'Alerie',
+    children: ['Margaery', 'Loras'],
+    house: 'Tyrell',
+  },
+  {
+    name: 'Sansa',
+    spouse: 'Tyrion',
+    house: 'Stark',
+  },
+  {
+    name: 'Jon',
+    spouse: null,
+    house: 'Snow',
+  },
+];
+
+const countNumberOfChildren = (arr) => {
+    // solution is here
+    const numOfChild = arr.reduce( (acc,num) =>
+    {
+      if (num.children.length != 0)
```

Comment:
I can tell that you and Nawal are working together, because the code style (braces) is non-standard. Please give credit to your partner when working together.

Reply:
i am sorry for forget tell you mr john , the other code challenge i will try to do alone and if i do it with nawal i will tell you